pkg/system/system.go

```diff
@@ -716,6 +726,7 @@ func Connect(usePortForwarding bool) (*Client, error) {
 		NooBaa:      r.NooBaa,
 		NBClient:    nbClient,
 		S3Client:    s3.New(s3Session),
+		S3DnsURL:    s3DnsURL,
```

Comment:
It is not clear that this is the internal DNS.
I would suggest to refrain from doing this logic inside Connect().
The provisioner can simply do this on its own.

Reply:
fixed in PR #179